web/src/components/Exchange/index.js

```diff
@@ -255,6 +255,10 @@ const ExchangePage = ({ points, setPage }) => {
 
   useEffect(() => {
     if (targetCoupon === null) return;
+    if (user.points - targetCoupon.points < 0) {
+      alert("The balance is not enough!");
+      return;
+    }
```

Comment:
optional: use toast?

Reply:
Looks good. We could use `toast` as the error handler.